client/lib/lib.rs

```diff
@@ -93,23 +93,35 @@ pub fn make_deploy(
     deploy_params: DeployStrParams<'_>,
     session_params: SessionStrParams<'_>,
     payment_params: PaymentStrParams<'_>,
+    force: bool,
 ) -> Result<()> {
-    let output = deploy::output_or_stdout(none_if_empty(maybe_output_path)).map_err(|error| {
-        Error::IoError {
-            context: format!(
-                "unable to get file or stdout, provided '{:?}'",
-                maybe_output_path
-            ),
-            error,
-        }
+    let maybe_output_path = none_if_empty(maybe_output_path);
+    let path_exists = if let Some(p) = maybe_output_path {
+        PathBuf::from(p).exists()
+    } else {
+        false
+    };
+
+    let output = deploy::output_or_stdout(maybe_output_path).map_err(|error| Error::IoError {
+        context: format!(
+            "unable to get file or stdout, provided '{:?}'",
+            maybe_output_path
+        ),
+        error,
     })?;
 
-    Deploy::with_payment_and_session(
-        deploy_params.try_into()?,
-        payment_params.try_into()?,
-        session_params.try_into()?,
-    )?
-    .write_deploy(output)
+    match (path_exists, force) {
```

Comment:
Maybe change this `match` to an `if let (true, false) = (path_exists, force) { err } else { o/p }`

Reply:
I agree, the `if let (...)` will be cleaner and slightly more readable. The tradeoff is that we lose some of the logic encapsulated in the other three cases listed in the second match arm.